implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OTelCollectorIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+        .withExposedPorts(4318, 9090); // HTTP receiver, Prometheus exporter
+
+    @Test
+    void collectorShouldExportMetrics() throws Exception {
+        MeterRegistry registry = createOtlpMeterRegistryForContainer(container);
+        Counter.builder("test.counter").register(registry).increment(42);
+        Gauge.builder("test.gauge", () -> 12).register(registry);
+        Timer.builder("test.timer").register(registry).record(Duration.ofMillis(123));
+        DistributionSummary.builder("test.distributionsummary").register(registry).record(24);
+
+        await().atMost(Duration.ofSeconds(5))
+            .pollDelay(Duration.ofMillis(100))
+            .pollInterval(Duration.ofMillis(100))
+            .untilAsserted(() -> whenPrometheusScraped().then().statusCode(200).body(not(blankOrNullString())));
+
+        // @formatter:off
+        // tags can vary depending on where you run your tests:
+        //  - IDE: no telemetry_sdk_version tag
+        //  - Gradle: telemetry_sdk_version has the version number
+        whenPrometheusScraped().then().body(
+            containsString("{job=\"test\",service_name=\"test\",telemetry_sdk_language=\"java\",telemetry_sdk_name=\"io.micrometer\""),
+
+            matchesPattern("(?s)^.*test_counter\\{.+} 42\\n.*$"),
+            matchesPattern("(?s)^.*test_gauge\\{.+} 12\\n.*$"),
+
+            matchesPattern("(?s)^.*test_timer_count\\{.+} 1\\n.*$"),
+            matchesPattern("(?s)^.*test_timer_sum\\{.+} 123\\n.*$"),
+            matchesPattern("(?s)^.*test_timer_bucket\\{.+,le=\"\\+Inf\"} 1\\n.*$"),
+
+            matchesPattern("(?s)^.*test_distributionsummary_count\\{.+} 1\\n.*$"),
+            matchesPattern("(?s)^.*test_distributionsummary_sum\\{.+} 24\\n.*$"),
+            matchesPattern("(?s)^.*test_distributionsummary_bucket\\{.+,le=\"\\+Inf\"} 1\\n.*$")
+        );
+        // @formatter:on
+    }
+
+    private OtlpMeterRegistry createOtlpMeterRegistryForContainer(GenericContainer<?> container) throws Exception {
+        return withEnvironmentVariables("OTEL_SERVICE_NAME", "test")
+            .execute(() -> new OtlpMeterRegistry(createOtlpConfigForContainer(container), Clock.SYSTEM));
+    }
+
+    private OtlpConfig createOtlpConfigForContainer(GenericContainer<?> container) {
+        return new OtlpConfig() {
+            @Override
+            public String url() {
+                return String.format("http://%s:%d/v1/metrics", container.getHost(), container.getMappedPort(4318));
+            }
+
+            @Override
+            public Duration step() {
+                return Duration.ofSeconds(1);
+            }
+
+            @Override
+            public String get(String key) {
+                return null;
+            }
+        };
+    }
+
+    private Response whenPrometheusScraped() {
+        // @formatter:off
+        return given()
+            .port(container.getMappedPort(9090))
+            .accept(OPENMETRICS_100)
```

Comment:
When I tried running the collector locally, I was not able to get it to return OpenMetrics format even when sending this `Accept` header and configuring the prometheus exporter with `enable_open_metrics: true`. Debugging this test, it looks like the response is also not OpenMetrics format.

Reply:
Great catch, I did not even notice this. This is another bug in the collector.
If I send 1.0.0 (what I should), it does not work:
```
Accept: application/openmetrics-text; version=1.0.0; charset=utf-8
```
If I send 0.0.1 (what I should not), it works:
```
Accept: application/openmetrics-text; version=0.0.1; charset=utf-8
```

See details here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18913

Reply:
I fixed this by requesting 0.0.1 and also included a TODO item.